feat(Peer Grouping Authoring): Implement authoring popup

geoffreykwan opened the pull request:
You may want to test this along with the WISE-API branch ```issue-114-rename-peer-group-activity```.

You will need to change these project content field names in your project.json.
- In the project content change peerGroupActivities to peerGroupings
- In the component content change peerGroupActivityTag to peerGroupingTag

## Changes
- Implemented Peer Grouping authoring
- Renamed PeerGroupActivity to PeerGroupings

## Test
- Make Peer Chat and Show Group Work authoring can set the Peer Grouping Tag
- Make sure you can create, edit, delete Peer Groupings
- Make sure you can author the Peer Grouping Tag for all components in their advanced authoring popup
- Make sure Peer Chat in the VLE works like before

Closes #552

geoffreykwan committed:
feat(Peer Group Authoring): Implement authoring popup

Displaying and choosing groupings works.
Adding new groupings and editing groupings does not work yet.

#552

geoffreykwan committed:
feat(Peer Group Authoring): Implement Add New Grouping

#552

geoffreykwan committed:
chore(Peer Group Authoring): Make the field labels bold

Show 'Steps Used In: None' when a grouping is not used in any steps.

#552

geoffreykwan committed:
feat(Peer Group Authoring): Save and update grouping settings in backend

#552

geoffreykwan committed:
chore(Peer Group Authoring): Rename some files

#552

geoffreykwan committed:
chore(Peer Group Authoring): Change label to name

#552

geoffreykwan committed:
feat(Peer Group Authoring): Always show all details

Remove 'Details' button.

#552

geoffreykwan committed:
feat(Peer Group Authoring): Allow deleting when not used in any steps

#552

geoffreykwan committed:
feat(Peer Group Authoring): Initialize Peer Groupings in JSON if needed

#552

geoffreykwan committed:
feat(Peer Group Authoring): Show message when there are no Groupings

#552

geoffreykwan committed:
fix(Peer Group Authoring): Fix creating and updating when no run

#552

geoffreykwan committed:
fix(Peer Group Authoring): Fix problem when updating a Peer Grouping

Previously a Peer Grouping would be updatd even if the author did not
click the Update button. Now the Update button must be clicked to save
the changes.

#552

geoffreykwan committed:
chore(Peer Group Authoring): Rename variable

Rename peerGroupSetting to peerGroupSettings.

#552

geoffreykwan committed:
feat(Peer Chat Authoring): Display Grouping Logic Name

#552

hirokiterashima committed:
refactor(Peer Chat Authoring): Pass in value to dialog instead of wrapper object

geoffreykwan committed:
feat(Peer Group Authoring): Changed edit settings to a popup

#552

geoffreykwan committed:
Merge branch 'issue-552-peer-group-activity-authoring' of https://github.com/WISE-Community/WISE-Client into issue-552-peer-group-activity-authoring

geoffreykwan committed:
chore(Peer Group Authoring): Fix a typo and clean up some code

#552

geoffreykwan committed:
feat(Peer Group Authoring): Change create new grouping to a popup

#552

geoffreykwan committed:
feat(Peer Group Authoring): Improve Select Peer Grouping popup UI

Added dividers and moved Cancel and Save buttons to the bottom.

#552

geoffreykwan committed:
feat(Peer Group Authoring): Implement authoring component

#552

geoffreykwan committed:
feat(Show Group Work): Add select grouping logic authoring element

#552

geoffreykwan committed:
chore(Peer Group Authoring): Rename SelectPeerGrouping component

Renamed from SelectPeerGrouping to SelectPeerGroupingOption.

#552

geoffreykwan committed:
chore(Peer Group Authoring): Remove unnecessary files

#552

geoffreykwan committed:
chore(Peer Group Settings): Rename Peer Grouping to Peer Group Settings

#552

geoffreykwan committed:
chore(Peer Group Settings): Refactor logic for when there is no run

#552

geoffreykwan committed:
chore(Peer Group Settings): Rename peerGrouping variables

Rename from peerGrouping to peerGroupSettings.

#552

geoffreykwan committed:
chore(Peer Group Settings): Show Steps Used In: None in editing view

#552

geoffreykwan committed:
chore(Peer Grouping Authoring): Rename files to peer-grouping

Renamed files and folders from peer-group-settings to peer-grouping.
Functions and variable names still need to be renamed.

#552

geoffreykwan committed:
chore(Peer Grouping Authoring): Rename variables and functions

Renamed variables and functions to peer-grouping.
These things still need to be renamed
- project content field
- component content field
- end point urls

#552

geoffreykwan committed:
test(Peer Grouping Authoring): Add tests

#552

geoffreykwan committed:
chore(Peer Grouping): Rename Peer Group Activity to Peer Grouping

#552

geoffreykwan committed:
chore(Peer Grouping): Rename Peer Group Settings to Peer Grouping

#552

geoffreykwan committed:
Merge branch 'issue-370-create-peer-chat-component' into issue-552-peer-group-activity-authoring

geoffreykwan committed:
refactor(Peer Grouping): Make components use new Peer Grouping authoring

Also show "None Selected" when a Peer Grouping hasn't been selected.

#552

geoffreykwan committed:
refactor(Peer Chat): Remove Peer Grouping authoring from advanced view

#552

hirokiterashima committed:
Merge branch 'issue-370-create-peer-chat-component' into issue-552-peer-group-activity-authoring

hirokiterashima commented on src/assets/wise5/authoringTool/peer-grouping/create-new-peer-grouping-dialog/create-new-peer-grouping-dialog.component.ts:
```diff
@@ -0,0 +1,37 @@
+
+  cancel(): void {
+    this.dialogRef.close();
+  }
```
Move to parent class? (will necessitate moving dialogRef to parent class constructor)

hirokiterashima commented on src/assets/wise5/authoringTool/peer-grouping/edit-peer-grouping-dialog/edit-peer-grouping-dialog.component.ts:
```diff
@@ -0,0 +1,44 @@
+
+  cancel(): void {
+    this.dialogRef.close();
+  }
```
Move to parent class? (will necessitate moving dialogRef to parent class constructor)

hirokiterashima commented on src/assets/wise5/authoringTool/peer-grouping/edit-peer-grouping-dialog/edit-peer-grouping-dialog.component.ts:
```diff
@@ -0,0 +1,44 @@
+  stepsUsedIn: string[] = [];
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public peerGrouping: any,
```
```suggestion
    @Inject(MAT_DIALOG_DATA) public peerGrouping: PeerGrouping,
```

hirokiterashima commented on src/assets/wise5/authoringTool/peer-grouping/select-peer-grouping-dialog/select-peer-grouping-dialog.component.ts:
```diff
@@ -0,0 +1,54 @@
+
+  selectPeerGrouping(tag: string): void {
+    this.selectedTag = tag;
+  }
```
Should this function take in PeerGrouping object instead of string? So, instead of using this.selectedTag, use this.selectedPeerGrouping?

hirokiterashima commented on src/assets/wise5/authoringTool/peer-grouping/select-peer-grouping-dialog/select-peer-grouping-dialog.component.ts:
```diff
@@ -0,0 +1,54 @@
+  constructor(
+    private dialog: MatDialog,
+    private dialogRef: MatDialogRef<SelectPeerGroupingDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) private selectedTag: string,
```
```suggestion
    @Inject(MAT_DIALOG_DATA) private selectedPeerGrouping: PeerGrouping,
```

Should we be working with PeerGrouping objects in SelectPeerGroupingDialogComponent and the dialog components that are launched from this class?

hirokiterashima commented on src/assets/wise5/authoringTool/peer-grouping/select-peer-grouping-option/select-peer-grouping-option.component.ts:
```diff
@@ -0,0 +1,46 @@
+  }
+
+  select(): void {
+    this.selectEvent.emit(this.peerGrouping.tag);
```
Should this emit this.peerGrouping?

breity committed:
style(Peer Grouping Authoring): Update styles for Peer Grouping authoring
- Hide completion threshold settings until implemented
#552

hirokiterashima commented on src/assets/wise5/authoringTool/peer-grouping/author-peer-grouping-settings/author-peer-grouping.component.ts:
```diff
@@ -0,0 +1,15 @@
+import { PeerGrouping } from '../../../../../app/domain/peerGrouping';
+
+@Directive()
+export abstract class AuthorPeerGroupingComponent implements OnInit {
```
Rename to AuthorPeerGroupingDialogComponent?

hirokiterashima replied:
Rename to AuthorPeerGroupingDialogComponent?

geoffreykwan committed:
refactor(Peer Grouping Authoring): Clean up code and fix some issues

Removed the cancel and save buttons and replaced them with a close
button. Selecting a Peer Grouping now saves and updates immediately.

#552

breity committed:
style(Peer Grouping Authoring): Set dynamic dialog widths
#552

hirokiterashima committed:
refactor(Peer Grouping Authoring): Extract PeerGroupingLogic

- Show translated logic name (Random, Manual)
- Remove commented out template

#552

hirokiterashima committed:
test(Peer Grouping Authoring): Fix test

#552
